db/Pos/DB/Functions.hs

```diff
@@ -18,28 +20,47 @@ module Pos.DB.Functions
 
 import           Universum
 
-import qualified Data.ByteString  as BS (drop, isPrefixOf)
-import           Formatting       (bprint, builder, sformat, shown, stext, string, (%))
+import qualified Data.ByteString    as BS (drop, isPrefixOf)
+import           Formatting         (bprint, builder, int, sformat, shown, stext, string,
+                                     (%))
 
-import           Pos.Binary.Class (Bi, decodeFull, serialize')
-import           Pos.DB.Class     (DBIteratorClass (..), DBTag, IterType, MonadDB (..),
-                                   MonadDBRead (..))
-import           Pos.DB.Error     (DBError (DBMalformed))
-import           Pos.Util.Util    (maybeThrow)
+import           Pos.Binary.Class   (Bi, decodeFull, serialize')
+import           Pos.Core.Constants (dbSerializeVersion)
+import           Pos.DB.Class       (DBIteratorClass (..), DBTag, IterType, MonadDB (..),
+                                     MonadDBRead (..))
+import           Pos.DB.Error       (DBError (..))
+import           Pos.Util.Util      (maybeThrow)
 
 
 -- | Read serialized value associated with given key from pure DB.
-dbGetBi
+dbGetBiNoVersion
     :: forall v m.
        (Bi v, MonadDBRead m)
     => DBTag -> ByteString -> m (Maybe v)
+dbGetBiNoVersion tag key = do
+    bytes <- dbGet tag key
+    traverse (fmap snd . dbDecode . (ToDecodeValue key)) bytes
```

Comment:
I don't get what's going on here. You decode `(Word8, v)` and then throw-away `Word8`. This function should be the opposite of `dbPutBiNoVersion`. It should decode `v`.